src/java.base/share/classes/sun/launcher/resources/launcher.properties

```diff
@@ -232,6 +232,21 @@ The following options are macOS specific:\n\
 \    -Xdock:icon=<path to icon file>\n\
 \                      override default icon displayed in dock\n\n
 
+# Translators please note do not translate the options themselves
+java.launcher.opt.concise.header  =   Usage: java [java options...] <application> [application arguments...]\n\n\
+\Where <application> is one of:\n\
+\  <mainclass>                to execute the main method of a compiled class\n\
+\  -jar <jarfile>.jar         to execute the main class of a JAR archive\n\
+\  -m <module>[/<mainclass>]  to execute the main class of a module\n\
+\  <sourcefile>.java          to compile and execute a source-file program\n\n\
```

Comment:
I'm not sure about the description of <mainclass>. It uses "compiled class", maybe you means "compiled main class" or something else to connect it <mainclass>?

"-jar <jarfile>.jar" may be confusing because the "java -help" uses "-java <jarfile">". I think the usages need to be the same.

Reply:
Used `a compiled main class` and `<jarfile>` (without `.jar`) as suggested in:
https://github.com/openjdk/jdk/pull/21411/commits/e9e9ad4bdd0a11244cb1ef84fdeea036e94718c7

Reply:
@AlanBateman I'd like to respectfully disagree that this was an improvement. If consistency was the important objection here, maybe it would have been better to update "java -help" to include the `.jar`?

In particular, losing the `.java` extension is problematic. I believe that this is essential to the launcher in how it determines if I am trying to launch a class or a source-file program. I just recently tried renaming my `HelloWorld.java` to `HelloWorld.java.old`, but when I ran `java HelloWorld.java.old` I got a warning implying that java thought I was trying to run the class `old` in the package `HelloWorld.java`.

I guess a better solution for the launcher would be to check if the argument given could match an existing text file, even if it looks like a fully qualified class name (but that does not exist). In fact, that might probably be worth a bug report of its own.

But I insist that as long as the current behavior stands, where the trailing `.java` is essential for the launcher to understand that it is launching a source-file program, then the `.java` literal should be a part of the help.

And, for the sake of consistency, that would also imply that the `.jar` suffix should be kept. (I don't know if the launcher would accept like `-jar foo.jar.old`..?)

I still recall how I **struggled** when learning Java that I should do `javac HelloWorld**.java**` but not `java HelloWorld**.class**`. By explicitly stating `.jar` and `.java` where those are expected, we indirectly underscore the important fact that the class specifications should be made **without** any extension.

Reply:
> @AlanBateman I'd like to respectfully disagree that this was an improvement. If consistency was the important objection here, maybe it would have been better to update "java -help" to include the .jar?

I assume your disagreement is with Jan's latest update rather than my comment. My comment was just pointing out that it's confusing to use `-jar <jarfile>.jar` in one case and `-jar <jarfile>` in the other.

Reply:
Done here:
https://github.com/openjdk/jdk/pull/21411/commits/d4860961e2ff7e6218a16a2ed3990a0deba3ae1a

Reply:
Thanks. Can you also post a snippet of `java -help` in the summary that shows how your change there looks?